packages/fetch/src/modifiers.ts

```diff
@@ -0,0 +1,146 @@
+import { createSignal, ResourceFetcherInfo } from "solid-js";
+import { RequestContext } from "./fetch";
+
+export type RequestModifier = <T>(...args: any[]) => (requestContext: RequestContext<T>) => any;
+
+export type Fetcher<T> = Exclude<RequestContext<T>['fetcher'], undefined>;
+
+export const wrapFetcher = <T>(
+  requestContext: RequestContext<T>,
+  wrapper: (originalFetcher: Fetcher<T>) => Fetcher<T>
+) => {
+  const originalFetcher = requestContext.fetcher
+  if (!originalFetcher) {
+    throw new Error("could not read resource fetcher");
+  }
+  requestContext.fetcher = wrapper(originalFetcher) || originalFetcher;
+};
+
+export const wrapResource = <T>(
+  requestContext: RequestContext<T>,
+  wrapper: (requestContext: RequestContext<T>) => [
+    props?: { [key: string]: any },
+    actions?: { [key: string]: any }
```

Comment:
I don't like the types here, they do not guide the user to provide correct values.
```suggestion
    props?: PropertyDescriptorMap,
    actions?: Record<PropertyKey, (...args: unknown[]) => unknown>
```
These would be better.
Also on another thought... if the props are being assigned via `Object.defineProperties`, then:
1. You have to use this lengthy syntax to have the getter assignad.
```ts
{
   aborted: { get: () => my_aborted_value },
   status: { get: () => my_status_value },
   response: { get: () => my_response_value }
}
```
2. Users can easily assign setters as well, which breaks the read/write separation. Actions would be for that.

So maybe the type for props should be:
```ts
Record<PropertyKey, Accessor<unknown>>
```
Used like this:
```ts
{
   aborted: () => my_aborted_value,
   status: () => my_status_value,
   response: () => my_response_value
}
```
And assigned like this:
```ts
for (const prop in props) Object.defineProperty(ctx, prop, { get: props[prop] })
```

Reply:
Actually, the types are still on my TODO list. I wanted to find a way to make sure the return value type of createFetch can be changed by the modifiers.

Reply:
Ah right. well, that sounds like fun :D
It should work similarly to mergeProps actually.

Reply:
yes, but the difficulty is that we have no return type to the modifiers, so I need to create some utility types.

Reply:
Why not have a `createModifier` then, that would provide the `wrapRequest` and `wrapFetcher` with types limited to stuff passed to generics?

Reply:
I don't want to add code just to satiate the type checker, so I want to try without first.